rtgui/thumbnail.cc

```diff
@@ -136,50 +135,37 @@ void Thumbnail::_generateThumbnailImage ()
     cfs.exifValid = false;
     cfs.timeValid = false;
 
-    if (ext == "jpg" || ext == "jpeg") {
-        infoFromImage (fname);
-        tpp = rtengine::Thumbnail::loadFromImage (fname, tw, th, -1, pparams->wb.equal);
-
-        if (tpp) {
-            cfs.format = FT_Jpeg;
-        }
-    } else if (ext == "png") {
-        tpp = rtengine::Thumbnail::loadFromImage (fname, tw, th, -1, pparams->wb.equal);
-
-        if (tpp) {
-            cfs.format = FT_Png;
-        }
-    } else if (ext == "tif" || ext == "tiff") {
-        infoFromImage (fname);
-        tpp = rtengine::Thumbnail::loadFromImage (fname, tw, th, -1, pparams->wb.equal);
+    // RAW works like this:
+    //  1. if we are here it's because we aren't in the cache so load the JPG
+    //     image out of the RAW. Mark as "quick".
+    //  2. if we don't find that then just grab the real image.
+    bool quick = false;
+    rtengine::RawMetaDataLocation ri;
+
+    rtengine::eSensorType sensorType = rtengine::ST_NONE;
+    if ( initial_ && options.internalThumbIfUntouched) {
+        quick = true;
+        tpp = rtengine::Thumbnail::loadQuickFromRaw (fname, ri, sensorType, tw, th, 1, TRUE);
+    }
 
-        if (tpp) {
-            cfs.format = FT_Tiff;
-        }
-    } else {
-        // RAW works like this:
-        //  1. if we are here it's because we aren't in the cache so load the JPG
-        //     image out of the RAW. Mark as "quick".
-        //  2. if we don't find that then just grab the real image.
-        bool quick = false;
-        rtengine::RawMetaDataLocation ri;
-
-        rtengine::eSensorType sensorType = rtengine::ST_NONE;
-        if ( initial_ && options.internalThumbIfUntouched) {
-            quick = true;
-            tpp = rtengine::Thumbnail::loadQuickFromRaw (fname, ri, sensorType, tw, th, 1, TRUE);
-        }
+    if ( tpp == nullptr ) {
+        quick = false;
+        tpp = rtengine::Thumbnail::loadFromRaw (fname, ri, sensorType, tw, th, 1, pparams->wb.equal, TRUE);
+    }
```

Comment:
@xiota FYI this may be causing some jpg files to open incorrectly because they are treated as raw files. See https://discuss.pixls.us/t/rt-problem-nikon-z8-w-z50mm-1-8s-lens/43933

Reply:
Thanks.  Will look at more.

Reply:
I have a potential fix.  Do you want it in a separate PR?  Would it be okay to add to #7096, along with a couple other misc changes, like a workflow update that needs to be in dev to make testing libjxl 0.7.x easier.